bot/seasons/easter/egg_decorating.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import random
+from io import BytesIO
+from pathlib import Path
+
+
+import discord
+from PIL import Image
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+COLOURS = [
+    (255, 0, 0, 255), (255, 128, 0, 255), (255, 255, 0, 255), (0, 255, 0, 255),
+    (0, 255, 255, 255), (0, 0, 255, 255), (255, 0, 255, 255), (128, 0, 128, 255)
+]  # Colours to be replaced - Red, Orange, Yellow, Green, Light Blue, Dark Blue, Pink, Purple
+
+IRREPLACEABLE = [
+    (0, 0, 0, 0), (0, 0, 0, 255)
+]  # Colours that are meant to stay the same - Transparent and Black
+
+
+class EggDecorating(commands.Cog):
+    """A Command that lets you decorate some easter eggs!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(aliases=["decorateegg"])
+    async def eggdecorate(self, ctx, *colours):
+        """
+        This 'paints' a beautiful egg using inputted colours
+
+        It picks from a random set of designs and alters the colours to the user's liking
+        """
+
+        if len(colours) < 2:
+            return await ctx.send("You must include at least 2 colours!")
+
+        invalid = []
+        converted = []
+        for c in colours:
+            try:
+                colour = await commands.ColourConverter().convert(ctx, c)
+                # Attempts to convert the arguments into discord.Colour
+                converted.append(colour)
+            except commands.BadArgument:
+                invalid.append(c)
+
+        if len(invalid) > 1:
+            return await ctx.send(f"The following colours are invalid: {' '.join(invalid)}")
+        elif len(invalid) == 1:
+            return await ctx.send(f"{invalid[0]} is an invalid colour!")
+
+        colours = converted
+
+        async with ctx.typing():
+            colours *= 4
+            # This is to ensure that no IndexErrors are raised since the most amount of colours on an egg is 8
+            num = random.randint(1, 6)
+            im = Image.open(Path("bot", "resources", "easter", "easter_eggs", f"design{num}.png"))
+            data = list(im.getdata())
+
+            replaceable = {x for x in data if x not in IRREPLACEABLE}  # Turns it into a set to avoid duplicates
```

Comment:
Using a set here is self-explanatory so the comment isn't necessary.
```suggestion
            replaceable = {x for x in data if x not in IRREPLACEABLE}
```